### PR TITLE
Added link checker workflow

### LINKED ISSUE #193: Add link checker workflow
The goal is verify all the links in Readme are valid.

Refer to the workflow [here](https://github.com/cleanlab/cleanlab/blob/master/.github/workflows/links.yml)

### PR DESCRIPTION
**Pull request by manulpatel**
Added a link checker in the github workflow to check invalid links.

Closes #193

### CODE REVIEW

File: .github/workflows/links.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - run: |
+          find . -name '*.md' -exec pandoc -i {} -o {}.html \;
+      - uses: anishathalye/proof-html@v1
+        with:
+          directory: .
+          check_favicon: false
+          empty_alt_ignore: true
+          url_ignore_re: |
+            ^https:\/\/docs\.github\.com\/
+            ^https:\/\/github\.com\/cleanlab\/cleanlab\/projects
```
Comment (sanjanag) [inline .github/workflows/links.yml:27]:
```
^https:\/\/docs\.github\.com\/
^https:\/\/github\.com\/cleanlab\/cleanlab\/projects
```
can remove these 2 lines, they are related to other project

**Comment by sanjanag** [comment]
@manulpatel How did you test the workflow?

**Comment by manulpatel** [comment]
@sanjanag I couldn't. I think you need to first merge then only we could test.

**Comment by sanjanag** [comment]
@manulpatel You can test this on your own fork, try raising the PR or a push event on your fork and that should trigger this workflow.
For ensuring the check is working, try changing one of the links in readme to a dummy link that doesn't work and ensure that the worfklow fails.
The workflow should work in the other case reporting no broken links.

File: .github/workflows/links.yml (hunk 1 above)
Comment (manulpatel) [reply .github/workflows/links.yml:27]: Hmm Noted.

**Comment by manulpatel** [comment]
[Here ](https://github.com/manulpatel/cleanvision/actions/runs/5137015691/jobs/9244524922) are the results of the testing.  The errors arise due to missing `alt `attribute in images link.

**Comment by manulpatel** [comment]
@sanjanag What do you suggest for **discussions** link? Should we remove it completely or are we planning to open discussions for cleanvision also?

**Comment by sanjanag** [comment]
@manulpatel I've updated the discussions link to direct to Slack now. Please update your branch with the main branch and all links should work then

**Review by sanjanag — APPROVED** [review]

**Pull request merged**